site-gen/src/main.rs

```diff
@@ -212,7 +212,11 @@ fn main() {
     let tags: BTreeSet<_> = tags
         .into_iter()
         .map(|tag| Tag {
-            posts: posts.clone(),
+            posts: posts
+                .clone()
+                .into_iter()
+                .filter(|p| p.tags.iter().find(|x| x.tag == tag.tag).is_some())
```

Comment:
This _could_ be an intersection if we chose to represent this as a BTreeSet (wouldn't need to explicitly sort, either).